api/controllers/resource_controller.js

```diff
@@ -11,131 +11,214 @@
+            status = 400;
         } else {
-            message = "Resource added successfully.";
+            message = `There was an unknown error while adding ${req.body.machine_name} as a resource.`;
+            status = 500;
         }
+        errors.push(err);
+    } finally {
         res.status(status).json({
-            success: !err,
-            error: err ? err : null,
+            success: (errors.length === 0),
+            errors: errors,
             message: message,
-            resource: new_resource ? new_resource : null,
+            data: new_resource,
         });
-    });
+    }
 };
 
-exports.update_resource_by_customer_id = (req, res) => {
-    res.status(501).json({
-        success: true,
-        error: null,
-        message: "NOT IMPLEMENTED",
-    });
+exports.update_resource_by_customer_id = async (req, res) => {
+    let status = 200;
+    let message = "";
+    let errors = [];
+    let updated_resource = null;
+
+    try {
+        // `req.body.update` should be a json encoded object with the fields to update
+        req.body.update = JSON.parse(req.body.update);
+    } catch (e) {
+        // Assume this is valid json; if not mongoose will throw an error and we'll return it below
+    }
+    
+    try {
+        updated_resource = await Resources.findOneAndUpdate({_id: req.user_id},
+            // Unpack `update` and set `updatedOn` field.
+            {
+                ...req.body.update,
+                updatedOn: Date.now()
+            },
+            // Return the updated document with `new: true`
+            {
+                new: true,
+                runValidators: true,
+            }
+        );
+        message = `This resource was successfully updated.`;
+
+    } catch (err) {
+        status = 500;
+        message = `There was an error updating this resource.`;
+        errors.push(err)
+    } finally {
+        res.status(status).json({
+            // If there were no errors we'll say this was successful
+            success: (errors.length === 0),
+            errors: errors,
+            message: message,
+            data: updated_resource,
+        });
+    }
+
 };
 
-exports.delete_resource_by_id = (req, res) => {
+exports.delete_resource_by_id = async (req, res) => {
     let message;
     let status = 200;
-    let id = req.user_id;
+    let errors = [];
+    let deleted_resource = null;
 
-    Resources.remove({
-        owner: id,
-        _id: req.params.resource_id,
-    }, (err) => {
-        if(err) {
-            message = `There was an error while deleting the resource: ${req.body.resource_id}.\nError: ${err.name}`;
-            status = 500;
-        } else {
-            message = "Resource deleted successfully";
-        }
+    // Check for resource_id in params; value in body is deprecated
+    // See: https://github.com/deepmarket/api/wiki/API-Enpoints#resources-resources for documentation on this
+    let resource_id = req.params.resource_id;
```

Comment:
maybe add validation here? If there's no resource_id attached to the req then we don't need to have an async call to Resources.  If there's some check earlier that I missed, ignore this comment